### DISCUSSION TITLE
FXDQ50TV1B(A)

### DISCUSSION THREAD

**Discussion by valkenrath** [revk/xESP32-Faikout#153] [Q&A]
Having scoured README/issues/discussions, still not sure whether my 2x FXDQ-TV1B(A) in-wall units can make use of this awesome project.

As best I can tell, no S21 port on those units, but an X50A port *does* exist (albeit sans cable).

Pics: '[wiring diagram](https://share.breakglass.io/Lesu2/mozIpuSo28.jpg/raw)' from the device in question (not convinced it's 1:1 accurate), and what I hope is the [X50A](https://share.breakglass.io/Lesu2/tEgoXije92.jpg/raw) data port.

Questions:

1. Are these HVAC units compatible with Faikin control? Is any more info required to elucidate compatibility?
2. Assuming yes to above, how would one power the device, as I'm of the understanding the X50A port is just data, not power. 
3. Best way to source necessary cable(s)? Best guess at present is 'JST ZH' cable for the X50A port, and 'JST XH' for power. 

Thanks!

**Comment by revk** [comment]
My understanding is that there is a separate connector for power. If nobody else comes along to explain, I'll have to see if I can get in to my loft and get pictures.

**Reply by revk** [reply]
Ah, I have found some pictures. This is how my unit is wired to `X50A` and `X35A`.

![IMG_0463](https://github.com/revk/ESP32-Faikin/assets/996983/f00f3f0b-ef9a-4d22-a3aa-3435d871b259)
![IMG_0464](https://github.com/revk/ESP32-Faikin/assets/996983/e7006755-96a2-4026-8835-c8b1ec5e685a)

**Reply by revk** [reply]
I have updated the main `README.md` to include these images now.

**Comment by valkenrath** [comment]
Awesome, thanks for the response.
Assuming I can locate an `X35A` connector on the controller, will be ordering some of your AmazUK boards today.

With regard to cabling, it seems I need exactly the same as your pictures depict - split data/power for the `X50A` / `X35A`, coalescing into a single five pin connector for the Faikin. Are you able to point me in the direction of a off-the-shelf SKU for that cable, or is DIY the only way forward?

Thanks again :)

**Reply by revk** [reply]
I don't have the exact spec of the connectors - someone else may. Amazon should have stock in a day or two.

**Comment by valkenrath** [comment]
Ordered 2x as soon as they were stocked, arrived recently.
Sourced cabling too, looks identical to your above pictures.

Alas, not plug and play :( By default, I get 'System is off line', and MQTT messages that seemed like it was switching between S21 and X50 (evidently to no avail). Found a thread where you'd helped remedy that error using `protofix`, so attempted to do the same. Below unsuccessful results, each with an AC reboot (at the fuse box) between.
To rule out hardware failure, I've also tried the second Faikin, with a different cable (harness?), and get the same: 'System is off line'.

```
{"ts":"2023-12-17T23:40:41Z","url":"http:\/\/ota.revk.uk\/Faikin-S3-MINI-N4-R2.bin","version":"18fcd76","project":"Faikin","time":"12:31:43","date":"Dec  5 2023","up-to-date":true}

protocol 0, protofix = true
comms → {"protocol":"S21","cmd":"F1","noack":true,"value":"FF"}
control → {"field":"power","error":"Setting cannot be controlled"}

protocol 1, protofix = true
comms → {"protocol":"S21¬Tx","cmd":"F1","noack":true,"value":"FE"}
control → {"field":"power","error":"Setting cannot be controlled"}

protocol 2, protofix = true
comms → {"protocol":"S21¬Rx","cmd":"F1","noack":true,"value":"01"}
control → {"field":"power","error":"Setting cannot be controlled"}

protocol 3, protofix = true
comms → {"protocol":"S21¬Tx¬Rx","cmd":"F1","noack":true,"value":"00"}
control → {"field":"power","error":"Setting cannot be controlled"}

protocol 4, protofix = true
comms → {"protocol":"X50A","badhead":true,"mismatch":true,"badrxlen":true,"data":"15EA"}

protocol 5, force = true
comms → {"protocol":"X50A¬Tx","badhead":true,"mismatch":true,"badrxlen":true,"data":"15EA"}

protocol 6, protofix = true
comms → {"protocol":"X50A¬Rx","badsum":"7A","data":"7505"}

protocol 7, protofix = true
comms → {"protocol":"X50A¬Tx¬Rx","badsum":"7A","data":"7505"}
```

Hoping for a hint as to where I've gone wrong.

Thanks!

**Reply by revk** [reply]
Do you have pictures of the wiring and the plugs at the aircon. That will help narrow it down. Also picture of the Faikin. With that we should be able to fix the exact protocol.

Actually, looking at that I suspect you are `S21`, so set `protofix` to `true`, and set `protocol` to `0`, and set `dump` to `true`. That will give more info.

But I suspect it will confirm the Daikin is not seeing the messages we send, which comes back to wiring, so pictures will help.

**Comment by valkenrath** [comment]
With `{ "protocol": 0, "protofix": true, "debug": true }`:
```
info/living-faikin/s21 → {}
```

With `{ "protocol": 4, "protofix": true, "debug": true }`:
```
info/living-faikin/cannot-tx → {"protocol":"X50A","cmd":"CB","payload":"0000"}
info/living-faikin/tx → {"protocol":"X50A","cmd":"AA","payload":"01"}
```

![cable](https://github.com/revk/ESP32-Faikin/assets/45926793/1c3d7ee6-785e-4046-a807-cc6db8875de1)

![installed](https://github.com/revk/ESP32-Faikin/assets/45926793/ee09058a-8625-40bf-9ec9-9dde4460eb2f)

**Reply by revk** [reply]
OK protocol `4` is definitely what you want.

You saw no `"rx"` messages at all? The previous logs showed some Rx errors.

I'll have to go in to me loft and check my X50A wiring to compare.

Do you have a meter you can check voltages on the Faikin connector?

Also the `cannot-tx` message is just that it has not seen rx messages, so won't send yet. It may be worth my changing the code so that with fixproto it sends anyway. I can look at that later this morning. (OK, done now).

Also, assuming you can do it without breaking your nice wiring, lifting the tab slightly you should be able to pull a wire out of the 5 pin connector, the brown one at the end (bottom of your first picture). Can you try that?

Also, I have made a code change to `"tx"` regardless if `protofix` is set.

**Comment by valkenrath** [comment]
- pulled new firmware OTA, didnt seem to make a difference
- (subsequently) pulled the brown wire out of the faikin-side connector, then rebooted AC for good measure

logs:
```
error/comms → 
{"protocol":"X50A","badhead":true,"mismatch":true,"badrxlen":true,"data":"15EA"}
{"protocol":"X50A","badsum":"FD","data":"15EA15EA15EA"}

info/tx → 
{"protocol":"X50A","cmd":"AA","payload":"01"}
{"protocol":"X50A","cmd":"CB","payload":"0000"}
{"protocol":"X50A","cmd":"CA","payload":"0000000000000000000000000000000000"}
```

**Reply by revk** [reply]
That is weird. I was really hoping the brown wire.

15EA15EA15EA is weird. That almost seems like a bad baud rate.

I'm slightly at a loss.

I'd be playing with an oscilloscope at this point. So not sure what to suggest.

**Reply by valkenrath** [reply]
Don't have metering equipment, but if it would likely bear fruit, could obtain some.

Is there specific info/data I could capture that aid in your diagnosis, or do I need {electronics knowledge and self-directed curiosity} to proceed further?

Either way, thanks for your assistance / attention thus far =)

**Reply by revk** [reply]
I really am a at a bit off a loss. Happy to do refund through Amazon, but would be better if we make it work.

Basic cheap volt meter may be useful to confirm the pins are at expected levels.

It looks like my wiring matches exactly.

![IMG_4236](https://github.com/revk/ESP32-Faikin/assets/996983/ac2e520b-bdfa-4264-accf-dc1211425ac6)
![IMG_4350](https://github.com/revk/ESP32-Faikin/assets/996983/c83c3f4c-7c6b-4e6c-a044-f446eed6b25e)
![IMG_4237](https://github.com/revk/ESP32-Faikin/assets/996983/d70cd789-c747-4d14-a29e-40e5d7eb4e9b)

**Reply by valkenrath** [reply]
For science, and to rule out bad HVAC controller, installed my other Faikin into the other (much harder to access) Daikin AC. 
It too has X50A/X35A ports - its the same device family, just different max throughput. 


On first power, pulled the latest FW OTA → `Faikin-S3-MINI-N4-R2: 791e522 2023-12-20T16:00:53`
Also set `protofix` to false, figuring your autodiscovery logic is better than any hardcoded value, given I've cycled through all of them above to no avail.

Results for posterity:
```
info/tx → {"protocol":"X50A","cmd":"AA","payload":"01"}
info/cannot-tx → {"protocol":"X50A","cmd":"CA","payload":"0000000000000000000000000000000000"}
info/cannot-tx → {"protocol":"X50A","cmd":"CB","payload":"0000"}
error/comms → {"protocol":"X50A","badsum":"F4","data":"75057505"}

info/tx → {"protocol":"X50A¬Tx¬Rx","cmd":"AA","payload":"01"}
info/cannot-tx → {"protocol":"X50A¬Tx¬Rx","cmd":"CB","payload":"0000"}
error/comms → {"protocol":"X50A¬Tx¬Rx","badhead":true,"mismatch":true,"badrxlen":true,"data":"15EA"}
```
Realised I was using the modified cable from prior experiments, so then switched it with a new, third cable (not used in any prior tests), with the brown pin intact. As best I can tell, eyeballing the MQTT messages and comparing them to those listed in this update, same outcome. 

So, no joy, but have at least ruled out physical fault with:
- Daikin AC (have tried two)
- Faikin ESP's (have tried two)
- cable fault (have tried three)

Comparing the results to prior experiments, I see error/comms response `7505` is different to the previous `15EA`, but watching the MQTT messages, it seems 15EA is associated with X50A *Rx*, and 7505 with *Tx*, so I'm guessing just a sampling miss, not a different result.

I'll look into sourcing a multimeter 👍🏻

**Reply by revk** [reply]
Thanks. The only other option is getting one of the official Daikin modules and confirming it *does* work.

As I say, if we get nowhere, they I can sort a refund.

**Reply by alexcf** [reply]
I have the FXQD32 and mine also doesn't respond. I've tried multiple different cables and no joy. I did notice on both the wiring diagram and in the docs that there is no mention of X50A or the support for the WiFi module.

I was hoping the Altherma_Schange might fix it as someone else had the same responses were getting. @revk i noticed there wasn't anything further on that thread either. 

@valkenrath Did you find a solution?
